src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -602,6 +594,36 @@ class IntervalExpressionSegment(BaseSegment):
 )
 
 
+class ObjectReferenceSegment(ansi.ObjectReferenceSegment):
+    """A reference to an object.
+
+    https://dev.mysql.com/doc/refman/8.0/en/account-names.html
+    https://dev.mysql.com/doc/refman/8.0/en/role-names.html
+    """
+
+    match_grammar: Matchable = OneOf(
+        ansi.ObjectReferenceSegment.match_grammar,
+        Sequence(
+            OneOf(
+                Ref("NakedIdentifierSegment"),
+                Ref("QuotedIdentifierSegment"),
+                Ref("SingleQuotedIdentifierSegment"),
+                Ref("DoubleQuotedLiteralSegment"),
+            ),
+            Sequence(
+                Ref("AtSignLiteralSegment"),
+                OneOf(
+                    Ref("NakedIdentifierSegment"),
+                    Ref("QuotedIdentifierSegment"),
+                    Ref("SingleQuotedIdentifierSegment"),
+                    Ref("DoubleQuotedLiteralSegment"),
+                ),
+                optional=True,
+            ),
```

Comment:
I presume should be no spaces? So might wanna make that obvious:

```suggestion
                optional=True,
                allow_gaps=False,
            ),
            allow_gaps=False,
```

Reply:
Ummmm .... it's ok with the first but not the second 🙃  However, probably good to just enforce no gaps anyways.


```mysql
mysql> GRANT INSERT, UPDATE, DELETE, SELECT, REFERENCES ON prj_table TO `prj_svc` @`%`;
Query OK, 0 rows affected (0.00 sec)

mysql> GRANT INSERT, UPDATE, DELETE, SELECT, REFERENCES ON prj_table TO `prj_svc` @ `%`;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '`%`' at line 1
```

Reply:
OK than can just enforce the first one and not the second (note they are in reverse order to your example):

```suggestion
                optional=True,
                allow_gaps=False,
            ),
```